packages/block-editor/src/hooks/duotone.js

```diff
@@ -304,28 +306,69 @@ function BlockDuotoneStyles( { name, duotoneStyle, id } ) {
  */
 const withDuotoneStyles = createHigherOrderComponent(
 	( BlockListBlock ) => ( props ) => {
-		const duotoneSupport = getBlockSupport(
-			props.name,
-			'color.__experimentalDuotone'
-		);
-
-		const id = `wp-duotone-${ useInstanceId( BlockListBlock ) }`;
-		const className = duotoneSupport
-			? classnames( props?.className, id )
+		const id = useInstanceId( BlockListBlock );
+
+		const selector = useMemo( () => {
+			const blockType = getBlockType( props.name );
+
+			if ( blockType ) {
+				// Backwards compatibility for color.__experimentalDuotone. This will
+				// have priority over filter.duotone support. Unfortunately we can't
+				// prefer filter.duotone because it gets set when __experimentalDuotone
+				// is set via a block_type_metadata_settings hook. It shouldn't be too
+				// much of a problem because I would expect consumers to not use both
+				// at the same time.
```

Comment:
I had to do this to fix the experimental selectors in the editor being applied to the root of the block instead of the selector set in __experimentalSelector.